sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -671,5 +671,6 @@ def create_scope_from_url(url):
 
 def validate_cache_staleness_value(max_integrated_cache_staleness):
     int(max_integrated_cache_staleness)  # Will throw error if data type cant be converted to int
-    if max_integrated_cache_staleness <= 0:
-        raise ValueError("Parameter 'max_integrated_cache_staleness_in_ms' can only be a positive integer.")
+    if max_integrated_cache_staleness < 0:
+        raise ValueError("Parameter 'max_integrated_cache_staleness_in_ms' can only be an "
+                         "integer greater than or equal to zero")
```

Comment:
Why are we doing client-side validation here instead of letting the service throw an error?

Reply:
I believe a negative value still gets accepted by the service but gets ignored altogether - this way the user knows the ranges of numbers they can provide
@JericHunter can you please remove this validation and double check that this is the case?

Reply:
If the service provides a good error, we should just remove this CS validation. If indeed it does simply ignore it, then this is good.

Reply:
The service does not provide a good error so we should keep this